src/h5web/visualizations/line/LineVis.tsx

```diff
@@ -27,36 +33,66 @@ function LineVis(props: Props): ReactElement {
     curveType = CurveType.LineOnly,
     showGrid = true,
     scaleType = ScaleType.Linear,
+    abscissas,
   } = props;
 
-  const indexDomain = extendDomain([0, dataArray.size - 1], 0.01);
-  const dataDomain = useMemo(() => {
-    return domain
-      ? extendDomain(domain, 0.05, scaleType === ScaleType.Log)
+  const abscissaValues = abscissas || range(dataArray.size);
+  const abscissaToIndex = getValueToIndexScale(
+    abscissaValues.map(
+      (x, i) =>
+        abscissaValues[i - 1] + (abscissaValues[i] - abscissaValues[i - 1]) / 2
+    )
+  ); // Shifting index change threshold to midpoints
```

Comment:
Note that you could use `x` instead of `abscissaValues[i]`.

One thing that I find a bit obscure is the fact that the shifting is done in the caller, but the slicing is done in `getValueToIndexScale`. The function does a bit of the work, but not all of it, creating an implicit coupling with the caller.

I feel like the shifting should be done inside `getValueToIndexScale`, perhaps with a boolean argument like `shiftThresholds` and a comment explaining when to use it.

Reply:
> Note that you could use `x` instead of `abscissaValues[i]`.

I found the syntax `abscissaValues[i] - abscissaValues[i - 1]` clearer than `x - abscissaValues[i - 1]`

> One thing that I find a bit obscure is the fact that the shifting is done in the caller, but the slicing is done in `getValueToIndexScale`. The function does a bit of the work, but not all of it, creating an implicit coupling with the caller.
> 
> I feel like the shifting should be done inside `getValueToIndexScale`, perhaps with a boolean argument like `shiftThresholds` and a comment explaining when to use it.

Yes I see your point, I will try to implement this